File: src/main/java/Funding/Startreum/domain/reward/exception/RewardExceptionHandler.java
package Funding.Startreum.domain.reward.exception;

import Funding.Startreum.common.util.ApiResponse;
import Funding.Startreum.domain.reward.controller.RewardRestController;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice(assignableTypes = RewardRestController.class)
public class RewardExceptionHandler {

    @ExceptionHandler({EntityNotFoundException.class,})
    public ResponseEntity<ApiResponse<Void>> handleException(EntityNotFoundException e) {
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(ApiResponse.error(e.getMessage()));
    }

}

File: src/main/java/funding/startreum/domain/reward/exception/RewardExceptionHandler.kt
package funding.startreum.domain.reward.exception

import funding.startreum.common.util.ApiResponse
import funding.startreum.domain.reward.controller.RewardRestController
import jakarta.persistence.EntityNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice(assignableTypes = [RewardRestController::class])
class RewardExceptionHandler {

    @ExceptionHandler(EntityNotFoundException::class)
    fun handleException(e: EntityNotFoundException): ResponseEntity<ApiResponse<Void>> {
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
            .body(ApiResponse.error(e.message!!))
    }
}
